core/src/main/java/org/eclipse/hono/config/MapperEndpoint.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.config;
+
+import java.util.Objects;
+
+/**
+ * Configuration class for custom mappers.
+ */
+public class MapperEndpoint {
+    private String host;
+    private Integer port;
+    private String uri;
+
+    /**
+     * Gets the host name or IP address of this mapper.
+     *
+     * @return The host name.
+     */
+    public String getHost() {
+        return host;
+    }
+
+    /**
+     * Sets the host name or IP address of this mapper.
+     *
+     * @param host The host name or IP address.
+     * @throws NullPointerException if host is {@code null}.
+     */
+    public void setHost(final String host) {
```

Comment:
how about making this class immutable and provide a static factory method that accepts all three properties?

Reply:
Doesn't work with the spring boot properties binding.